hypothesis-python/src/hypothesis/internal/reflection.py

```diff
@@ -112,6 +109,12 @@ def function_digest(function):
         hasher.update(function._hypothesis_internal_add_digest)
     except AttributeError:
         pass
+    # We add the qualified name to distinguish e.g. identical test methods of
+    # subclasses that call into different overridden methods.
+    try:
+        hasher.update(function.__qualname__.encode())
+    except AttributeError:
+        pass
```

Comment:
Unfortunately this won't help if the method is inherited, because qualname shows the defining class.

The case where users explicitly define _literally identical_ test functions on independent classes might be rare enough that it's not worth invalidating everyones databases over it 🤔.  We could leave commented-out code to be enabled when next we break it though?

Reply:
Good call, I didn't consider the invalidation problem. The benefit is marginal as you say. I can put a comment in. Is there a greppable comment marker I should use?

Reply:
No special marker, but this is basically the only code that could globally invalidate databases so we're very likely to see it anyway (and I'll remember too).